src/scripts/main.js

```diff
@@ -1,3 +1,30 @@
 'use strict';
 
-// write code here
+document.addEventListener('DOMContentLoaded', () => {
+  const tree = document.querySelector('.tree');
+
+  // Додаємо слухач подій
+  tree.addEventListener('click', (e) => {
+    const target = e.target;
+
+    if (target.tagName === 'SPAN') {
+      const parentLi = target.closest('li');
+
+      if (!parentLi) {
+        return;
+      }
+
+      const childUl = parentLi.querySelector('ul');
+
+      if (childUl) {
+        childUl.hidden = !childUl.hidden;
+      }
+    }
+  });
+
+  const nestedLists = tree.querySelectorAll('ul');
```

Comment:
Make sure that the `tree` variable is not `null` before calling `querySelectorAll` on it. If `tree` is `null`, this will throw an error.